src/compiler/commandLineParser.ts

```diff
@@ -974,6 +974,8 @@ namespace ts {
         {
             name: "keyofStringsOnly",
             type: "boolean",
+            affectsSemanticDiagnostics: true,
+            affectsEmit: true,

```

Comment:
Actually this flag doesn't affect JS emit. It only affects declaration emit. In fact anything that changes type inference (like strictNullChecks, noImplicitAny) affects semantic diagnostics and declaration emit.
Is it worth adding a new flag `affectsTypeInference` that implicitly contains `affectsSemanticDiagnostics` and `affectsEmit` only for declaration files?